### ISSUE TITLE
Trampoline classes causes compiler warning: violates the C++ One Definition Rule  [-Wodr]

### ISSUE THREAD

**Issue by v-yatsenko** [RosettaCommons/binder#51]
Assume we have following hierarchy:
```
namespace n1 {
  class my_exception : public std::exception {}
  namespace n2 {
    class my_exception: public ::my_exception {}
  }
}
```
cppbinder will generate a trampoline structures:

```
struct PyCallBack_my_exception : public n1::my_exception {}
```
```
struct PyCallBack_my_exception : public n1::n2::my_exception {}
```
And even worse if situation will be the same as in this example. There will be an error because of type redefinition.
In our project we have such types in separate modules so we have only warnings.

**Comment by v-yatsenko** [comment]
As a solution propose following change:
In source/class.cpp:
```
// generate call-back structure name for given class
inline string callback_structure_name(CXXRecordDecl const *C)
{
        string ret = "PyCallBack_";
        ret += replace_(C->getQualifiedNameAsString(), "::", "_");
        return ret;
}
```

**Comment by lyskov** [comment]
Good catch @v-yatsenko ! This should be fixed in 89aa2c6bb9a550be299a0e3cf91d423eb6a23c27
